### ISSUE TITLE
Add option for always adding quotes around types

### ISSUE THREAD

**Issue by jaredkhan** [dropbox/mypy-protobuf#282]
One of my main reasons for enjoying mypy-protobuf, beyond Mypy type-checking, is improving the IDE experience of working with protobuf generated code. Me and many of my colleagues use PyCharm, which seems to have a bug to do with doubly-nested classes in stub files. https://youtrack.jetbrains.com/issue/PY-50517. When using mypy-protobuf, this PyCharm bug means that autocompletion doesn't work in certain situations.

For example:
```
syntax = "proto3";

message Message1 {
  message Message2 {
    message Message3 {
      string my_string = 1;
    }
    Message3 my_submessage = 1;
  }
}
```

```
Message1.Message2().my_submessage.
```
auto-completion from this point should suggest 'my_string' but doesn't.

To work around this PyCharm bug, it would be nice if the mypy-protobuf generated stubs had the option of always quoting the message types. For example
```
        @property
        def my_submessage(self) -> global___Message1.Message2.Message3: ...
```
would become:
```
        @property
        def my_submessage(self) -> "global___Message1.Message2.Message3": ...
```

which seems to fix this inconvenience.

**Comment by nipunn1313** [comment]
Hi - thanks for providing an example.
It does seem like pycharm is not able to understand forward references (eg referencing Class1 before it is done being defined). This kind of thing is very typical inside type checkers (for example a recursive data structure must by typed this way).

Stealing your example from the linked issue, I was able to repro the pycharm issue with just this
```
    class Class1:
        class Class2:
            a: str = ...
        prop: Class1.Class2
```

In our case, we're not actually using a recursive data structure. I believe pycharm won't trip if we show
```
    class Class1:
        class Class2:
            a: str = ...
        prop: Class2
```

We could update mypy-protobuf to do this instead of the fully qualified name which it currently shows (Class1.Class2). I would be willing to support that change in mypy-protobuf.

However, I don't want to support flags to regress unnecessarily to quotation syntax. Flags end up having a maintenance cost, and I'd rather have that energy apply pressure to work forward (make fixes in pycharm) rather than backward (quotation fwd references). I do appreciate the idea though.

**Comment by jaredkhan** [comment]
Just to make sure we're on the same page: the PyCharm bug, as far as I understand it, is specifically to do with stub files.
The smaller example you provided:

```
class Class1:
        class Class2:
            a: str = ...
        prop: Class1.Class2
```

works as I would expect in a stubfile for me (using a Python 3.7 interpreter). Autocompletion works correctly. Does it cause issue in a stubfile for you?

I think PyCharm's handling of forward references in non-stub files is okay, i.e. it raises errors as appropriate if you reference a class before its defined, whether in a type-hint or otherwise, unless either the type annotation is a string or `from __future__ import annotations` is used.

I agree with the fix you suggest though, that seems to work for me both in the toy example and the real-world examples it was motivated by.

If you think that's straightforward then that seems like a good solution, but it does look like that PyCharm ticket has already been triaged and reproduced on their end so perhaps, although it's useful to have discussed it, it's also sensible to just wait 2 weeks and see if this problem disappears ✨

**Comment by nipunn1313** [comment]
In a regular .py file it doesn't work
![image](https://user-images.githubusercontent.com/1300387/131508125-382abe5e-81f7-429e-935b-ac143a701165.png)
![image](https://user-images.githubusercontent.com/1300387/131508085-ea9cb117-4c10-43a9-937b-4054984bca68.png)


In a stub file, it works with one level of nesting, but not with two levels of nesting
Definitely something weird with forward references on type annotations going on for them
![image](https://user-images.githubusercontent.com/1300387/131508247-df904a8f-e5f4-498b-a994-d48b464a9e1c.png)

These work with mypy
https://mypy-play.net/?mypy=latest&python=3.10&gist=875ffc94f0b87295b65d21411327bc70

**Comment by nipunn1313** [comment]
I wrote up an impl to do such relative references - but it doesn't actually always work. The mypy-protobuf test cases actually caught this case
https://mypy-play.net/?mypy=latest&python=3.10&gist=553a13b0f942a4de35e0dc9decbebcec

Here
```python
class Class1:
    class Class2:
        a: str

        class Class3:
            a: str

        b: Class3
        c: Class2
        d: Class1.Class2.Class3
        e: Class1.Class2
```

b, d, e all work. But c does not work. Referencing Class2 at that scope requires notation in e, not like c. Confirmed both with python runtime and mypy.

Given this, we probably can't avoid this pycharm bug w/ the workaround I suggested.

Going to close this - encourage working towards a fix forward on pycharm side. You're welcome to fork mypy-protobuf to add quotes in the short term, but I don't think mypy-protobuf should afford maintaining such a mode. I wrote up #284 to show how it might be done.
